addon/ng2/utilities/fs-promise.ts

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+import * as fs from 'fs';
+
+/**
+ * Read a file and return a promise
+ * @param path (path to file)
+ * @return Promise with file content
+ */
+export function readWithPromise(path: string): Promise<any> {
+    var content: string;
+    return new Promise<any>((resolve, reject) => {
+      fs.readFile(path, (err, data) => {
```

Comment:
It's probably safe to assume this will be used to read files in utf8 format, which would let you simplify this a little bit.

- `fs.readFile(path, 'utf8', (...))`
- `data` would then be a string - no need to call `.toString()` on it
- Return type can be `Promise<string>`.